components/Pagination.js

```diff
@@ -0,0 +1,148 @@
+import { useEffect, useState } from "react";
+import { FaChevronRight } from "react-icons/fa";
+import { FaChevronLeft } from "react-icons/fa";
+import { BsChevronDoubleRight } from "react-icons/bs";
+import { BsChevronDoubleLeft } from "react-icons/bs";
+
+const Pagination = ({
+  data,
+  currentPage,
+  paginate,
+  perPage,
+  startIndex,
+  endIndex,
+}) => {
+  const [screenWidth, setScreenWidth] = useState();
+  const totalResults = data.length;
+  const totalPages = Math.ceil(totalResults / perPage);
+
+  useEffect(() => {
+    setScreenWidth(window.innerWidth);
+  }, []);
+
+  useEffect(() => {
+    // Add keydown event listener to handle pagination
+    const handleKeyDown = (e) => {
+      if (e.key === "ArrowLeft" && currentPage > 1) {
+        paginate(currentPage - 1);
+      }
+      if (e.key === "ArrowRight" && currentPage < totalPages) {
+        paginate(currentPage + 1);
+      }
+    };
+    window.addEventListener("keydown", handleKeyDown);
+
+    // Clean up the event listener on component unmount
+    return () => {
+      window.removeEventListener("keydown", handleKeyDown);
+    };
+  }, [currentPage, totalPages, paginate]);
+
+  let beforePage = currentPage - 1;
+  let afterPage = currentPage + 1;
+  const MEDIUM_SCREEN = 768;
+  const LARGE_SCREEN = 1024;
+  const pageNumber = [];
+
+  // medium screen pagination layout
+  if (screenWidth >= MEDIUM_SCREEN && screenWidth < LARGE_SCREEN) {
+    beforePage = currentPage - 2;
+    afterPage = currentPage + 2;
+  }
+
+  // large screen pagination layout
+  if (screenWidth >= LARGE_SCREEN) {
+    beforePage = currentPage - 3;
+    afterPage = currentPage + 3;
+  }
```

Comment:
I could not see this have an effect from the screenshot...? I am not sure this is needed as both looked fine

SMALLEST

<img width="491" alt="Screenshot 2023-07-07 at 05 56 15" src="https://github.com/EddieHubCommunity/LinkFree/assets/624760/e1295cce-a49b-4db7-ac20-bdfe3fde332e">


LARGEST

<img width="1586" alt="Screenshot 2023-07-07 at 05 56 22" src="https://github.com/EddieHubCommunity/LinkFree/assets/624760/3cda3e37-0ca9-4ac4-93a1-edcd72f9a6bd">

Reply:
On a large screen, this will display three page numbers (ex:  2 3 4 **5** 6 7 8) before and after the current page with next/prev and jump to first/last button when the current page >4

on a medium screen, this will display two page numbers (ex: 2 3 **4** 5 6) before and after the current page with next/prev button when the current page > 3

on a small screen, it uses the default `beforePage` and `afterPage`, this will display (2 **3** 4) one page number before and after the current page with next/prev button

Reply:
From my screenshots you will see I couldn't replicate the page numbers reducing - was I doing something wrong?

Reply:
You just need navigate different page and see the effects

Reply:
I always get the same amount of pagination options no matter the screen size. In your example screenshots the amount of pages display gets smaller as the screen does

LARGE

<img width="2541" alt="Screenshot 2023-07-09 at 17 59 14" src="https://github.com/EddieHubCommunity/LinkFree/assets/624760/1ffd7fd0-131e-48cd-833a-e0c8a865beaf">

SMALL

<img width="555" alt="Screenshot 2023-07-09 at 17 59 07" src="https://github.com/EddieHubCommunity/LinkFree/assets/624760/1b0c9cd9-92b9-44ea-ba88-48079ace4edc">

Reply:
I think the only issue remaining is the number of pages displayed does not change as the page size changes - maybe this logic could be removed if it is blocking the PR and seems to work fine without it?

Reply:
Ok, I will remove this logic 👍

Reply:
I removed that logic👍

Reply:
Thank you so much for the amazing feature and being consistent to push it forwards 💪  I look forward to more geeking out again soon!